src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -269,31 +254,41 @@ private void initCustomStyleOptions() {
   private void initSeedField() {
     seedTextField.setText(String.valueOf(generatorPrefs.getSeed()));
     generatorPrefs.seedProperty().bind(seedTextField.textProperty());
-    seedTextField.disableProperty()
-                 .bind(previousMapName.textProperty()
-                                      .isNotEmpty()
-                                      .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
-                                      .or(commandLineArgsText.textProperty().isNotEmpty())
-                                      .or(fixedSeedCheckBox.selectedProperty().not()));
-    seedRerollButton.disableProperty()
-                    .bind(previousMapName.textProperty()
-                                         .isNotEmpty()
-                                         .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
-                                         .or(commandLineArgsText.textProperty().isNotEmpty())
-                                         .or(fixedSeedCheckBox.selectedProperty().not()));
-  }
-
-  private void bindCustomStyleDisabledProperty(Node... nodes) {
+    seedTextField.disableProperty().bind(getExtendedFixedSeedNotBooleanBinding());
+    seedRerollButton.disableProperty().bind(getExtendedFixedSeedNotBooleanBinding());
+  }
+
+  private BooleanBinding getSharedBooleanBinding() {
+    return previousMapName.textProperty().isNotEmpty()
+                          .or(generationTypeComboBox.valueProperty().isNotEqualTo(GenerationType.CASUAL))
+                          .or(commandLineArgsText.textProperty().isNotEmpty());
+  }
+
+  private BooleanBinding getExtendedFixedSeedNotBooleanBinding() {
+    return getSharedBooleanBinding()
+        .or(fixedSeedCheckBox.selectedProperty().not());
+  }
+
+  private BooleanBinding getExtendedCustomStyleNotBooleanBinding() {
+    return getSharedBooleanBinding()
+        .or(customStyleCheckBox.selectedProperty().not());
+  }
```

Comment:
I would probably just inline this function as well even if it is used twice since I don't know if there is really a clear name for it and inlining would be easier to understand